### ISSUE TITLE
Support of OWS metadata in process definition

### ISSUE THREAD

**Issue by cehbrecht** [geopython/pywps#189]
In the current pywps-4 implementation one can provide metadata (for process, inputs and outputs) only as a list of strings. These strings become then OWS metadata elements with a `title` attribute.

The OWS metadata element has also the `href` attribute. I extensively use the metadata to provide links to external docs to enhance the process documentation (users need to understand what the process does :)

I provide a patch where one can provide a list of tuples `[(title, href), ]` for metadata in addtion to a list of strings. But i don't know how you would like to have it ...

**Comment by tomkralidis** [comment]
cc @jachym @ldesousa 

@cehbrecht thanks for the report and associated PR in #190.

IMHO it's a good time to change the Process model to make `metadata` a list of dicts, i.e.:

```python
...
metadata=[
    {
        'title': 'foo process',
        'href': 'http://example.org/foo,
    },
    {
        'title': 'foo process more documentation',
        'href': 'http://example.org/bar'
    }
]
...
```

..thus following the OWS metadata model from OWS 1.1.0.

Thoughts?

**Comment by tomkralidis** [comment]
Sample diff:

```diff
diff --git a/pywps/app/Process.py b/pywps/app/Process.py
index e52e746..bfc2ea5 100644
--- a/pywps/app/Process.py
+++ b/pywps/app/Process.py
@@ -112,7 +112,14 @@ class Process(object):
             doc.append(OWS.Abstract(self.abstract))
 
         for m in self.metadata:
-            doc.append(OWS.Metadata({'{http://www.w3.org/1999/xlink}title': m}))
+            metadata = {}
+            if 'href' in m:
+                metadata['{http://www.w3.org/1999/xlink}href'] = m['href']
+            if 'title' in m:
+                metadata['{http://www.w3.org/1999/xlink}title'] = m['title']
+            if 'type' in m:
+                metadata['{http://www.w3.org/1999/xlink}type'] = m['type']
+            doc.append(OWS.Metadata(metadata))
 
         for p in self.profile:
             doc.append(WPS.Profile(p))
```

**Comment by ldesousa** [comment]
According to the specification there can indeed be more than one metadata reference element. But it is defined in another document, I guess that is why no one ever complained about it.

I would prefer to have a Metadata class instead of a list of assorted composition. Does it exist in OWSLib? 

Also, the '{http://www.w3.org/1999/xlink}' string should be promoted to a property to avoid replication.

**Comment by tomkralidis** [comment]
@ldesousa there is no Metadata class in OWSLib (note OWSLib is mainly for parsing, not serializing).  having `pywps.app.Process.metadata` as a class would require a bit more from users when setting up their processes as opposed to passing a simple dict, but we already expect that given `pywps.app.Process.inputs`.

Having said this, how about a `Metadata` class in `pywps.app.Process`?  So then:

```python
from pywps.app.Process import Metadata
...
metadata=[Metadata('foo process', 'http://example.org/foo'), Metadata('foo process more documentation', 'http://example.org/bar')]
 ```

**Comment by ldesousa** [comment]
Yes, it was something like that I had in mind. You know, Computer Scientists formalities, but that long run facilitate things.

**Comment by jonas-eberle** [comment]
+1 for this kind of Metadata class.

**Comment by tomkralidis** [comment]
cc @jachym 
FYI associated pull request in #195 for review/comment/merge.

**Comment by jachym** [comment]
Big thanks works for me

**Comment by tomkralidis** [comment]
Implemented in #195.
